db/controllers/controllers.js

```diff
@@ -28,26 +28,72 @@ exports.getAllArticles = (req,res) => {
 
 exports.getCommentsById = (req, res) => {
     const { article_id } = req.params
-    selectCommentById(article_id).then((comment)=>{
-      
-        if(!comment){
-            console.log(comment, "controller")
-            return res.status(404).send({msg: '404 not found'})
+    
+    selectArticleById(article_id).then((article)=>{
+        if(!article){
+            return res.status(404).send({ msg: 'Article not found' })
+
         }
+    })
+    selectCommentsByArticleId(article_id)
+                .then((comments) => {
+                    if (comments.length === 0) {
+                        return res.status(200).send({ msg: 'No comments found for this article' })}
 
-        
-        
-     
+                    res.status(200).send({ comments });
+                })
+            }
+    
+    
 
-        res.status(200).send({comment})
+
+ exports.postCommentToArticle = (req, res) => {
+    console.log(req.body)
+   const { article_id } = req.params
+   if(!req.body.username){
+    return res.status(400).send({msg: 'Username not present'})
+   }
+   else 
+   if(!req.body.body){
+    return res.status(400).send({msg: 'Body not present'})
+
+   }
+    const { username, body } = req.body
+    
+    addCommentToArticle(article_id, username, body).then((comment)=>{
+        console.log(comment, "controller")
+        return res.status(200).send({comment})
+    }).catch((error)=>{
+        console.log(error)
     })
+
  }
 
 
- exports.postCommentToArticle = (req, res) => {
+ exports.patchController = (req,res) => {
     const {article_id} = req.params
-    addCommentToArticle(article_id).then((comment)=>{
-        return res.status(200).send({comment})
-    })
+    const { inc_votes } = req.body
+    selectArticleById(article_id).then((article) => {
+        if (!article){
+            return res.status(400)
+            .send({msg: 'invalid, article id not present'}) }
+
+            else 
+            if(isNaN(inc_votes)){
+                return res.status(400)
+                .send({msg: 'inc_votes is not a number'})
+            }
+
```

Comment:
all of this logic could be handled with PSQL errors which would come back from your model.